### PR TITLE
[android] style: enhance theme to match the green theme of om

### PR DESCRIPTION
**Pull request by kavikhalique**
Fix: #7590 
**Enhanced the UI and Matched the theme of OM Android app with the theme of Organic map**
- The theme of android app of OM is very mismatching somewhere its sky blue and somewhere its green

**Issue Solved**
- Greenified the whole theme accoridng to OM
- Enhanced Night Mode Theme with theme of Organic green theme.

https://github.com/organicmaps/organicmaps/assets/120750626/6f1c8b60-4b17-48dd-b07b-f2ac236c03c2

### CODE REVIEW

**Comment by vng** [comment]
Thanks, looks fresh, IMO!

**Comment by Misalf-git** [comment]
But, please, not the map screen buttons and things that weren't blue before.

**Comment by kavikhalique** [comment]
> But, please, not the map screen buttons and things that weren't blue before.

Previous grey color was giving feeling of disabled and less interactive according to the actual greenish theme, i have just spread the green color little bit so that it provide a green organic feel.

**Comment by kavikhalique** [comment]
> That's way too much green imo. I would keep the colors of the buttons as they are.

Please suggest where do you feel it is too much greenish. I will try to make it better.

**Comment by Misalf-git** [comment]
These:

:--: | :--: | :--: | :--:
 |  |  |

**Comment by kavikhalique** [comment]
> That's way too much green imo. I would keep the color of the buttons as they are.

Ok, will try to make it look less greenish and more better.
Please suggest if you have any idea it will be very helpful.

**Comment by meenbeese** [comment]
@kavikhalique I appreciate you picking up my PR as I seriously lacked the motivation to continue. That said, there are serious contrast issues with the colors in dark theme imo. I think my previous PR did a better job with the colors. Also, the redesigned folder icons are subpar as they are too large now.

**Comment by biodranik** [comment]
Blue color meant something actionable/clickable. Now it mixes with the background, unclickable green.

**Comment by Misalf-git** [comment]
> Also, the redesigned folder icons are subpar as they are too large now.

Oh, I didn't notice the changed icons.
The "download-arrow" icon does look better / more familiar, though.

**Comment by kavikhalique** [comment]
> @kavikhalique I appreciate you picking up my PR as I seriously lacked the motivation to continue. That said, there are serious contrast issues with the colors in dark theme imo. I think my previous PR did a better job with the colors. Also, the redesigned folder icons are subpar as they are too large now.

Will try to do some better changes to make it look better. Thankyou for your suggestions.

**Comment by euf** [comment]
This is a welcome change! 

Do you mind if we tweak colors of some elements a little more? 
If that's okay, in a few days I'm gonna suggest separate colors to be used for…

- Splash screen background;
- Menu header background;
- Accent color for the menu buttons;
- Accent color for on-the map buttons.

Based on the brand book UI color scheme: https://github.com/organicmaps/organicmaps/wiki/Colors

Judging from your screenshots, these adjustments may further improve consistency and contrast.

**Comment by kavikhalique** [comment]
> This is a welcome change! 
> 
> 
> 
> Do you mind if we tweak colors of some elements a little more? 
> 
> If that's okay, in a few days I'm gonna suggest separate colors to be used for…
> 
> 
> 
> - Splash screen background;
> 
> - Menu header background;
> 
> - Accent color for the menu buttons;
> 
> - Accent color for on-the map buttons.
> 
> 
> 
> Based on the brand book UI color scheme: https://github.com/organicmaps/organicmaps/wiki/Colors
> 
> 
> 
> Judging from your screenshots, these adjustments may further improve consistency and contrast.

I would love to work on this.

**Comment by pastk** [comment]
> @kavikhalique I appreciate you picking up my PR as I seriously lacked the motivation to continue. That said, there are serious contrast issues with the colors in dark theme imo. I think my previous PR did a better job with the colors. 

Could you please link to your previous PR?

**Comment by pastk** [comment]
> * The theme of android app of OM is very mismatching somewhere its sky blue and somewhere its green

TBH it never jumped to my eye as some inconsistency :/ Maybe I've just used to it over the years, of course :)
I don't see an issue having a few different colors in the UI (as long they're used consistently, of course).

Maybe the currently used blue color is not the best choice for actionable controls, indeed. However to my taste now everything looks **too green and too monotonous**.

Looks like quite some users from the chat also prefer the current grey maps buttons:
image

Looking forward for more ideas as e.g. @euf suggested!

**Comment by kavikhalique** [comment]
> > * The theme of android app of OM is very mismatching somewhere its sky blue and somewhere its green
> 
> 
> 
> TBH it never jumped to my eye as some inconsistency :/ Maybe I've just used to it over the years, of course :)
> 
> I don't see an issue having a few different colors in the UI (as long they're used consistently, of course).
> 
> 
> 
> Maybe the currently used blue color is not the best choice for actionable controls, indeed. However to my taste now everything looks **too green and too monotonous**.
> 
> 
> 
> Looks like quite some users from the chat also prefer the current grey maps buttons:
> 
> image
> 
> 
> 
> Looking forward for more ideas as e.g. @euf suggested!
> 
> 
> 
> 

The first thing that strike my mind after opening the app was the colour. Although present theme is not bad but i feel there should be a systematic use of colours not only green but any combination of colours should be used in a proper combination and systematic way (as said by @euf as-well).

While using the app as a first time user i was unable to figure out the logic of combination of blue and green bcz they were just appearing randomly.

I am looking forward to contribute some positive changes to OM.

**Comment by biodranik** [comment]
> While using the app as a first time user i was unable to figure out the logic of combination of blue and green bcz they were just appearing randomly.

Where is it inconsistent in your opinion? Blue means something clickable (except main map and bottom toolbar buttons), green is usually a background color.

**Comment by kavikhalique** [comment]
> > While using the app as a first time user i was unable to figure out the logic of combination of blue and green bcz they were just appearing randomly.
> 
> Where is it inconsistent in your opinion? Blue means something clickable (except main map and bottom toolbar buttons), green is usually a background color.

**Here blue colour has been used for text theme and its no clickable**


**Here report a bug is clickable but its not blue**


**Here blue colour was just used to show enable and disable**


IMO instead of using blue as clickable we can use green as well. Whatsapp also have green theme and they use green for both clickable and disabled as well.
Blue can also be used in case of OM but at someplaces blue and green should have used in combination as well to give a feeling that both colors are systematically chosen.

**Comment by biodranik** [comment]
Those discrepancies should be carefully fixed. Mixing a green background with clickable green may not be too obvious for existing and new users.

Does it make sense to start correcting the mentioned issues instead of rewriting the whole style?

**Comment by kavikhalique** [comment]
> Those discrepancies should be carefully fixed. Mixing a green background with clickable green may not be too obvious for existing and new users.
> 
> Does it make sense to start correcting the mentioned issues instead of rewriting the whole style?

yes it can also be done. But if we are selecting dual colour theme we should bring blue colour in themes of OM as well IMO.

**Comment by kavikhalique** [comment]
> Those discrepancies should be carefully fixed. Mixing a green background with clickable green may not be too obvious for existing and new users.
> 
> 
> 
> Does it make sense to start correcting the mentioned issues instead of rewriting the whole style?

IMO here green one looks better i guess

image

previous one

image

**Comment by biodranik** [comment]
The Blue color is a familiar, attractive one. It is used in many other apps to attract user's attention to click the most important button. Check other maps apps for example.

**Comment by kavikhalique** [comment]
> If that's okay, in a few days I'm gonna suggest separate colors to be used for…
>
>
It will be very helpful if you guide me through it because at this point i am really very confused what should i choose.

In my opinion, We can do either of the three things 

1 - let the UI be as it is, just replace green at places where it is un clickable but blue in colour as suggested by @biodranik 

2 - redesign UI properly considering both green and blue as a part of the brand colour and select theme which gives effect of both colour

3 - implement a simple toggle button in preferences to let the users choose between either monotonic theme or green tone theme.

**Comment by biodranik** [comment]
Start with 1, and think about 2. Definitely not 3.

**Comment by euf** [comment]
I agree, definitely not **3**. 

At the moment I'm working on **2**, trying to figure out the newest "Material 3" design approach. For building color schemes it uses second and third accent colors in addition to a primary one. I was hoping that would simply translate to our green vs. blue approach, but things turned out to be more complicated. Further tinkering is needed, and now I'm not sure it's going to work out soon. 

This leaves us with **1** if we are ready to move incrementally. Do you mind sharing screenshots of this approach?

**Comment by kavikhalique** [comment]
> Start with 1, and think about 2. Definitely not 3.

Ok for now will start with 1 and with suggestions will improve it

File: android/app/src/main/res/color/accent_color_selector_night.xml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <selector xmlns:android="http://schemas.android.com/apk/res/android">
-  <item android:color="@color/base_accent_night"
+  <item android:color="@color/bg_night_primary"
         android:state_selected="true"/>
 
   <item android:color="@color/text_light_subtitle"/>
```
[the remark this answers is not in the document]
Comment (kavikhalique) [reply android/app/src/main/res/color/accent_color_selector_night.xml:6]: Sorry if i am understanding wrong but do you mean that we should not add .png files instead we should add other format or you mean we should not add any more image resources to the project.

[the remark this answers is not in the document]
Comment (RedAuburn) [reply android/app/src/main/res/color/accent_color_selector_night.xml:6]: > Sorry if i am understanding wrong but do you mean that we should not add .png files instead we should add other format or you mean we should not add any more image resources to the project.

i think he means it'd be better to use SVG 👍️

[the remark this answers is not in the document]
Comment (kavikhalique) [reply android/app/src/main/res/color/accent_color_selector_night.xml:6]: Oh thankyou, understood

**Comment by kavikhalique** [comment]
> What is the current status here? 

Since there are a lot of confusions. I have gave it a pause since various distinct opinions are coming which are opposite of counter of each others.

**Comment by Jean-BaptisteC** [comment]
I suggest for this PR to keep only changes in light theme.
Update dark theme colors and icons can be done later.
@kavikhalique can you rebase your branch?

**Comment by kavikhalique** [comment]
> I suggest for this PR to keep only changes in light theme.
> 
> Update dark theme colors and icons can be done later.
> 
> @kavikhalique can you rebase your branch and apply suggestions?

Sure i will do that

**Comment by kavikhalique** [comment]
- Replaced png icons with svg files
- Removed Green icons from bottom sheet
- Changed the night mode to skyblue

@Jean-BaptisteC 

Here is the video of UI


https://github.com/organicmaps/organicmaps/assets/120750626/d0bef130-8589-4827-8437-7fabcc43e89a

**Comment by Jean-BaptisteC** [comment]
I prefer buttons in grey

**Comment by euf** [comment]
I'm not 100% sure about the green accent on the map buttons, but let's try and see how this works IRL. 
Replacing PNGs with SVGs would definitely allow more flexibility in further adjustments of UI colors 👍

**Comment by Misalf-git** [comment]
For me, in the light style, color on the buttons make them less "readable". In the dark style they pop out too much.

**Comment by kavikhalique** [comment]
According to past few suggestions i think its better to revert back the button color to grey in light mode and white in dark mode.

@rtsisyk @Jean-BaptisteC @euf

**Comment by biodranik** [comment]
Changing important action colors in the app should be carefully planned and designed. It would be great to first design it properly and consistently with our long-term plans (e.g. new Place Page, new Navigation UI), and change colors only after that.

P.S. Compare the importance of this issue/change (and time spent on it) with some other issues, crashes, bugs, missing features, etc.

**Comment by biodranik** [comment]
My opinion:
1. There are a lot of more important issues to fix than this one. The app works fine without this change.
2. It would also be great to spend reviewers' time more efficiently, e.g. by focusing on the most impactful/critical issues.
3. The chosen color is controversial and not clear.
4. Changing colors is tricky and may break the UX/complicate the code
5. Changing colors now will highly likely require changing them again later after the app redesign. So it would be better to postpone it.

**Comment by kavikhalique** [comment]
After various reviews and suggestions, right now its not correct time to change the UI so closing this PR.

**Pull request closed without merging**